source/fundamentals/authentication/mechanisms.txt

```diff
@@ -162,6 +161,36 @@ authentication mechanism and credentials in the connection string:
 .. literalinclude:: /code-snippets/authentication/aws.js
    :language: javascript
 
+If you need to specify an AWS session token, you can add it using one of
+the following choices:
+
+- **Specify your AWS session token in a connection string.**
+
+  To specify your AWS session token in a connection string,
+  append your session token with the ``authMechanismProperties``
+  parameter using the format ``AWS_SESSION_TOKEN:<awsSessionToken>``.
+  Your code to append a session token should look something like this:
+
+  .. code-block:: javascript
+
+     // Replace the following with values for your environment.
+     const accessKeyId = encodeURIComponent("<AWS_ACCESS_KEY_ID>");
+     const secretAccessKey = encodeURIComponent("<AWS_SECRET_ACCESS_KEY>");
+     const clusterUrl = "<MongoDB cluster url>";
+
+     let uri =`mongodb+srv://${accessKeyId}:${secretAccessKey}@${clusterUrl}/?authMechanism="MONGODB-AWS"`;
+     
+     const sessionToken = encodeURIComponent("<AWS_SESSION_TOKEN>");
+     uri = uri.concat(`&authMechanismProperties=AWS_SESSION_TOKEN:${sessionToken}`);
```

Comment:
This seems odd to be to declare the URI then append the session token. Also, this can be automatically read if not passed in the uri.